### ISSUE TITLE
False positives for containers with role="presentation"

### ISSUE THREAD

**Issue by TrevorBurnham** [squizlabs/HTML_CodeSniffer#274]
Prompted by https://github.com/mui-org/material-ui/issues/18106

If you set `role="presentation"` on any container element whose descendants have a semantic meaning, you'll get the warning:

> This element's role is "presentation" but contains child elements with semantic meaning.

For example, this markup triggers the warning:

```html
<div role="presentation">
  <a href="http://example.com">Link</a>
</div>
```

The [sniff](https://github.com/squizlabs/HTML_CodeSniffer/blame/0be3ec30064dcec3054fda8edbede3c1128ad3f9/Standards/WCAG2AAA/Sniffs/Principle1/Guideline1_3/1_3_1.js#L99-L124) that generates the warning seems to be based on an overly broad interpretation of [F92](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F92):

> This failure occurs when a role of presentation is applied to an element *whose purpose is to convey information or relationships in the content.*

The example given is that a `<table>` element with `role="presentation"` no longer functions as a table. The important thing is the element itself, not its descendants. The [WAI-ARIA spec](https://www.w3.org/TR/wai-aria-practices-1.1/#presentation_role_effects) specifically states that when you set `role="presentation"` on an element,

> The roles, states, and properties of each descendant element remain visible to assistive technologies

---

I recommend that the sniff should be rewritten to apply more narrowly. At the very least, it should ignore elements like `div` and `span` that already have no semantic meaning, since setting `role="presentation"` on those elements has no effect on assistive technologies.

This came up previously as https://github.com/squizlabs/HTML_CodeSniffer/issues/191 in the context of `<svg>` elements. /cc @ironikart

**Comment by ironikart** [comment]
Thanks @TrevorBurnham. I'm marking this as a bug since I believe you are correct, the sniff is currently too broad and HTMLCS will show an error in cases where it isn't an actual failure. I'll look into improving this.

**Comment by ironikart** [comment]
I'm closing this issue off. The last update I made is as far as I'm able to reasonably take the checks and addresses the issue.

**Comment by oliviertassinari** [comment]
Awesome :)
